pkg/prometheus/promcfg.go

```diff
@@ -724,6 +724,48 @@ func (cg *ConfigGenerator) addProxyConfigtoYaml(
 	return cfg
 }
 
+func (cg *ConfigGenerator) addCustomHTTPConfigtoYaml(
+	cfg yaml.MapSlice,
+	store assets.StoreGetter,
+	customHTTPConfig monitoringv1.CustomHTTPConfig,
+) yaml.MapSlice {
+	if reflect.ValueOf(customHTTPConfig).IsZero() {
+		return cfg
+	}
+
+	if cg.version.LT(semver.MustParse("2.55.0")) {
```

Comment:
this situation should be prevented in resource_selector.go already?

Reply:
Yes. Just logger warn.
Remove it done.